data-prepper-plugins/key-value-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/keyvalue/KeyValueProcessor.java

```diff
@@ -155,12 +163,19 @@ private boolean validateRegex(final String pattern)
         return true;
     }
 
-    private void validateKeySets(final Set<String> includeSet, final Set<String> excludeSet) {
-        Set<String> intersectionSet = new HashSet<String>(includeSet);
-        intersectionSet.retainAll(excludeSet);
-        if (!intersectionSet.isEmpty()) {
+    private void validateKeySets(final Set<String> includeSet, final Set<String> excludeSet, final Set<String> defaultSet) {
+        final Set<String> includeIntersectionSet = new HashSet<String>(includeSet);
+        final Set<String> defaultIntersectionSet = new HashSet<String>(defaultSet);
+
+        includeIntersectionSet.retainAll(excludeSet);
+        if (!includeIntersectionSet.isEmpty()) {
             throw new IllegalArgumentException("Include keys and exclude keys set cannot have any overlap", null);
         }
+
+        defaultIntersectionSet.retainAll(excludeSet);
+        if (!defaultIntersectionSet.isEmpty()) {
+            throw new IllegalArgumentException("Cannot exclude a default key!", null);
```

Comment:
Nit: do we need the second null argument?

Reply:
just checked, we don't, but a lot of the other `IllegalArgumentException` elsewhere in the code has the null at the end